Mathlib/RingTheory/DedekindDomain/FiniteAdeleRing.lean

```diff
@@ -361,6 +383,17 @@ instance : CoeFun (FiniteAdeleRing R K)
     (fun _ ↦ ∀ (v : HeightOneSpectrum R), adicCompletion K v) where
   coe a v := a.1 v
 
+section Algebra
+
+variable (L : Type*) [Field L] [Algebra L K] [Algebra L (FiniteAdeleRing R K)]
+  [IsScalarTower L K (FiniteAdeleRing R K)]
+
+theorem algebraMap_extension (r : L) :
+  algebraMap L (FiniteAdeleRing R K) r = algebraMap K (FiniteAdeleRing R K) (algebraMap L K r) :=
+  IsScalarTower.algebraMap_apply L K (FiniteAdeleRing R K) r
```

Comment:
This theorem I am a bit confused about. We have it already -- as you note in your proof, it's called `IsScalarTower.algebraMap_apply`. Why do we want a new name for it? Furthermore I don't think `algebraMap_extension` is the correct name for this. Searching through mathlib, `_extension` seems to be used for extending a set to a bigger set, or extending a function from a dense subspace. Probably it should just be called `algebraMap_apply` again? But this just brings me back to the question about why we need it at all.

Reply:
As above I included it from your recommendation; but I also felt that it was maybe unnecessary. Perhaps in the linked PR you meant another statement as I admit my naming of have's was not the most clear and consistent?

But, I agree it is not needed, as I returned to my code I submitted for 269 on FLT and you can replace the have h with a rw [IsScalarTower...].